core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala

```diff
@@ -0,0 +1,297 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.kafka.controller
+
+import kafka.api.LeaderAndIsr
+import kafka.common.TopicAndPartition
+import kafka.controller.{KafkaController, LeaderIsrAndControllerEpoch}
+import kafka.server.{KafkaConfig, KafkaServer}
+import kafka.utils.{CoreUtils, TestUtils, ZkUtils}
+import kafka.zk.ZooKeeperTestHarness
+import org.junit.{After, Before, Test}
+
+class ControllerIntegrationTest extends ZooKeeperTestHarness {
+  var servers = Seq.empty[KafkaServer]
+
+  @Before
+  override def setUp() {
+    super.setUp
+    servers = Seq.empty[KafkaServer]
+  }
+
+  @After
+  override def tearDown() {
+    servers.foreach(_.shutdown())
+    servers.foreach(server => CoreUtils.delete(server.config.logDirs))
+    super.tearDown
+  }
+
+  @Test
+  def testEmptyCluster(): Unit = {
```

Comment:
This test seems redundant given testControllerEpochPersistsWhenAllBrokersDown()?